### PR TITLE
chore: Java PubSub Tests

### PR DESCRIPTION
**Pull request by stuartwdouglas**

### CODE REVIEW

File: backend/controller/pubsub/integration_test.go [hunk 1]
```diff
@@ -34,13 +35,14 @@ func TestPubSub(t *testing.T) {
 				WHERE
 					state = 'success'
 					AND origin = '%s'
-		`, dal.AsyncOriginPubSub{Subscription: schema.RefKey{Module: "subscriber", Name: "testSubscription"}}.String()),
+		`, dal.AsyncOriginPubSub{Subscription: schema.RefKey{Module: "subscriber", Name: "testTopicSubscription"}}.String()),
```
Comment (stuartwdouglas) [inline backend/controller/pubsub/integration_test.go:38]: I renamed this one so the Java test could use the auto generated subscription annotation, I left the other the same to allow the use of an explicit name.

File: backend/controller/pubsub/testdata/java/subscriber/src/main/java/xyz/block/ftl/java/test/subscriber/Subscriber.java [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    private static final AtomicInteger catchCount = new AtomicInteger();
+    @TestTopicSubscription
+    void consume(PubSubEvent event) throws Exception {
+        Log.infof("Subscriber is consuming %s", event.getTime());
+    }
+
+    @Subscription(
+            topic = "topic2",
+            module = "publisher",
+            name = "doomedSubscription"
```
Comment (alecthomas) [inline backend/controller/pubsub/testdata/java/subscriber/src/main/java/xyz/block/ftl/java/test/subscriber/Subscriber.java:27]: This should be typesafe ideally, is this just temporarily using string constants?

**Review by alecthomas — APPROVED** [review]

File: backend/controller/pubsub/testdata/java/subscriber/src/main/java/xyz/block/ftl/java/test/subscriber/Subscriber.java (hunk 2 above)
Comment (stuartwdouglas) [reply backend/controller/pubsub/testdata/java/subscriber/src/main/java/xyz/block/ftl/java/test/subscriber/Subscriber.java:27]: The one above is typesafe, using the generated `@TestTopicSubscription`. I didn't rename this one (as the generated code has no way of knowing I want my subscription to be named like this, and also to test out the non-generated approach).

If this use case is common I guess I could add a `@SubscriptionName` annotation you could combine to override the auto generated one.

Comment (alecthomas) [reply backend/controller/pubsub/testdata/java/subscriber/src/main/java/xyz/block/ftl/java/test/subscriber/Subscriber.java:27]: Aaah I see, this is kind of a low-level API.

Comment (stuartwdouglas) [reply backend/controller/pubsub/testdata/java/subscriber/src/main/java/xyz/block/ftl/java/test/subscriber/Subscriber.java:27]: Yea, it's also a meta-annotation, so the generated one looks like:

```
/**
 * Subscription to the topic of type {@link PubSubEvent}
 */
@Retention(java.lang.annotation.RetentionPolicy.RUNTIME)
@Subscription(
    topic = "topic2",
    module = "publisher",
    name = "topic2Subscription"
)
public @interface Topic2Subscription {
}
```

**Pull request merged**